src/inmanta/server/agentmanager.py

```diff
@@ -1160,6 +1172,78 @@ async def _ensure_agents(
                     LOGGER.warning("Not all agent instances started successfully")
                 return start_new_process
 
+    # Start/Restart scheduler
+    async def _ensure_scheduler(
```

Comment:
I wonder if we should refactor ensure_agents because we share the same logic except the following:
```
            agent_map: Mapping[str, str] = cast(
                Mapping[str, str], await env.get(data.AUTOSTART_AGENT_MAP, connection=connection)
            )  # we know the type of this map
            autostart_agents: Set[str] = set(agents) & agent_map.keys()
            if len(autostart_agents) == 0:
                return False
```

Reply:
Maybe not to make the removal of old code easier.